### PR TITLE
Improve bloodhound connector with Netbios domain name

### PR DESCRIPTION
**Pull request by NeffIsBack**
image

Most often the bloodhound connector can now handle it when only the netbios name of the domain is specified. 
Fixes issues like #60, #78

FYI: this is branched of #35 so it is already up-to-date if this one gets merged

### CODE REVIEW

File: nxc/helpers/bloodhound.py [hunk 1]
```diff
@@ -43,13 +43,21 @@ def add_user_bh(user, domain, logger, config):
         try:
             with driver.session() as session, session.begin_transaction() as tx:
                 for info in users_owned:
+                    distinguished_name = "".join(["DC=" + dc + "," for dc in info["domain"].split(".")]).rstrip(",")
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:46]: Should we use an f-string here?

File: nxc/helpers/bloodhound.py [hunk 2]
```diff
@@ -43,13 +43,21 @@ def add_user_bh(user, domain, logger, config):
...
+                    domain_query = tx.run(f'MATCH (d:Domain) WHERE d.distinguishedname STARTS WITH "{distinguished_name}" RETURN d').data()
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:47]: Does the inner quotes have to be double? I know in some cases that's a necessity due to the query language (or like with Windows sometimes).

File: nxc/helpers/bloodhound.py [hunk 3]
```diff
@@ -43,13 +43,21 @@ def add_user_bh(user, domain, logger, config):
...
+                    if not domain_query:
+                        raise Exception("Domain not found in bloodhound")
+                    else:
+                        domain = domain_query[0]["d"].get("name")
+
                     if info["username"][-1] == "$":
-                        user_owned = info["username"][:-1] + "." + info["domain"]
+                        user_owned = info["username"][:-1] + "." + domain
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:54]: f-string?

File: nxc/helpers/bloodhound.py [hunk 4]
```diff
@@ -43,13 +43,21 @@ def add_user_bh(user, domain, logger, config):
...
                         account_type = "Computer"
                     else:
-                        user_owned = info["username"] + "@" + info["domain"]
+                        user_owned = info["username"] + "@" + domain
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:57]: f-string?

File: nxc/helpers/bloodhound.py [hunk 5]
```diff
@@ -43,13 +43,21 @@ def add_user_bh(user, domain, logger, config):
...
                         account_type = "User"
 
+
                     result = tx.run(f'MATCH (c:{account_type} {{name:"{user_owned}"}}) RETURN c')
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:61]: Same quote question

File: nxc/helpers/bloodhound.py [hunk 6]
```diff
@@ -63,7 +71,10 @@ def add_user_bh(user, domain, logger, config):
             logger.fail(f"Neo4J does not seem to be available on {uri}.")
             return
         except Exception as e:
-            logger.fail(f"Unexpected error with Neo4J: {e}")
-            logger.fail("Account not found on the domain")
+            if "Domain not found in bloodhound" in str(e):
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:74]: Do you need to cast e to string?

File: nxc/helpers/bloodhound.py (hunk 1 above)
Comment (NeffIsBack) [reply nxc/helpers/bloodhound.py:46]: Good catch, absolutely!

File: nxc/helpers/bloodhound.py (hunk 2 above)
Comment (NeffIsBack) [reply nxc/helpers/bloodhound.py:47]: Nope, that was just Code with the old style which i copied lol. Will change that

File: nxc/helpers/bloodhound.py (hunk 6 above)
Comment (NeffIsBack) [reply nxc/helpers/bloodhound.py:74]: Yes, otherwise it crashes.

File: nxc/helpers/bloodhound.py [hunk 7]
```diff
@@ -43,26 +43,26 @@ def add_user_bh(user, domain, logger, config):
...
+                        user_owned = f"{info['username'][:-1]}.{domain}"
                         account_type = "Computer"
                     else:
-                        user_owned = info["username"] + "@" + domain
+                        user_owned = f"{info['username']}@{domain}"
                         account_type = "User"
 
 
-                    result = tx.run(f'MATCH (c:{account_type} {{name:"{user_owned}"}}) RETURN c')
+                    result = tx.run(f"MATCH (c:{account_type} {{name:'{user_owned}''}}) RETURN c")
```
Comment (Marshall-Hallenbeck) [inline nxc/helpers/bloodhound.py:61]: It looks like an extra single quote got caught in here? I don't have bloodhound set up in my lab, have you tested it?

Comment (NeffIsBack) [reply nxc/helpers/bloodhound.py:61]: Damn it, yes

Comment (NeffIsBack) [reply nxc/helpers/bloodhound.py]: Interestingly it still worked

**Review by Marshall-Hallenbeck — APPROVED** [review]
LGTM
I didn't test it but if you did it should be good @NeffIsBack 
I need to set up BH in my lab env

**Review by mpgn — APPROVED** [review]

**Comment by NeffIsBack** [comment]
Adding "reviewed" label as @Marshall-Hallenbeck reviewed the code already

**Review by mpgn — APPROVED** [review]

**Pull request merged**